src/common/src/array/struct_array.rs

```diff
@@ -219,6 +222,29 @@ impl Array for StructArray {
 }
 
 impl StructArray {
+    pub fn new(
```

Comment:
Having a public constructor seems go against the design of `ArrayBuilder`. We can make this private, but unsure if this is enough to prevent future abuse.